### PR TITLE
Fix `CSI Ps M` deleting lines above cursor

### LINKED ISSUE #2984: Don't delete lines above the cursor on CSI Ps M
`Alacritty` should clamp `Ps` to not go above cursor, so if you're on the bottom line, typing `echo -e '\e[1000;M'` shouldn't delete all the thing above and should be clamped to `1`.

P.s. All the other terminals clamp it(vte based, xterm, kitty).

### PR DESCRIPTION
**Pull request by chrisduerr**
Fixes #2984.

### CODE REVIEW

File: alacritty_terminal/src/term/mod.rs [hunk 1]
```diff
@@ -1564,9 +1564,12 @@ impl<T: EventListener> ansi::Handler for Term<T> {
...
     #[inline]
     fn delete_lines(&mut self, lines: Line) {
+        let origin = self.cursor.point.line;
+        let lines = min(self.lines() - origin, lines);
+
         trace!("Deleting {} lines", lines);
-        if self.scroll_region.contains(&self.cursor.point.line) {
-            let origin = self.cursor.point.line;
+
+        if self.scroll_region.contains(&self.cursor.point.line) && lines.0 > 0 {
```
Comment (kchibisov) [inline alacritty_terminal/src/term/mod.rs]: Could we swap checks?

**Comment by kchibisov** [comment]
I guess we need a regression test for this.

File: alacritty_terminal/src/term/mod.rs (hunk 1 above)
Comment (chrisduerr) [reply alacritty_terminal/src/term/mod.rs]: Probably.

**Comment by chrisduerr** [comment]
> I guess we need a regression test for this.

I'd say it makes sense to test both the clear behavior at the bottom and when lines actually get deleted. Should be possible to cover both with the same test.

**Pull request merged**